### ISSUE TITLE
MN Solr `collectionQuery` field query results & DataONE privileges

### ISSUE THREAD

**Issue by gothub** [NCEAS/metadig-engine#227]
One potential problem that could occur with the usage of the DataONE Solr `collectionQuery` field is that Solr can return different result sets for different DataONE users. 

The metadig-engine graphing program must create graphs of aggregated quality scores that match the set of DataONE pids that a portal has defined via their filters. The portal creator will be using their own DataONE identity, and metadig-engine will be using the MN admin identity to issue the query. These results can be different, for example, more pids could be retrieved via the MN admin token 

Any suggestions on how to resolve these two result sets are welcome.

**Comment by amoeba** [comment]
Couple of questions:

1. Just curious, what is the `collectionQuery` field for and what does its content look like?
2. So the engine does the querying/graphing, what permissions does it run with?
3. _Should_ someone be able to see scores for a collection they can't browse? i.e., if I can't even browse Portal X, can I see scores for its content?
4. _Should_ the scores/graphs for a given facet (e.g., collection/portal) differ depending on your DataONE subject? i.e., if part of Portal X is public read, another part is not public read, should a DataONE subject with full read access to the Portal X's content see different scores than, say, a public user?

**Comment by gothub** [comment]
@amoeba the `collectionQuery` field is added to the Solr index when a collection document is uploaded. The query is created by metacat-index and attempts to recreate the query that is performed by MetacatUI when a portal is displayed.

The metadig-engine graphing facility can access all pids.

Questions 3,4 bring up some interesting points, I'd like to hear what others have to say on this.
I think that the quality graphs should be based on the same set of pids that are viewable by the portal, but I'm not sure how to ensure this.

**Comment by gothub** [comment]
Regarding the remote Solr query using `collectionQuery`, possibly the client could construct the same identity filter that the MN query engine constructs. For example, when I am logged into to dev.nceas and view my test portal, the following query is constructed to view the portal related datasets:

```
{"responseHeader":{"status":0,"QTime":1600,"params":{"fq":"(readPermission:\"CN=arctic-data-admins,DC=dataone,DC=org\")OR(rightsHolder:\"CN=arctic-data-admins,DC=dataone,DC=org\")OR(readPermission:\"CN=knb-data-admins,DC=dataone,DC=org\")OR(rightsHolder:\"CN=knb-data-admins,DC=dataone,DC=org\")OR(readPermission:\"public\")OR(isPublic:true)OR(readPermission:\"http://orcid.org/0000-0002-2192-403X\")OR(rightsHolder:\"http://orcid.org/0000-0002-2192-403X\")OR(readPermission:\"authenticatedUser\")","q":"((text:*pelagic* AND originText:*\"Rachael Blake\"*)) AND (-obsoletedBy:* AND formatType:METADATA)","facet.limit":"-1","facet.field":["documents","id","geohash_2"],"fl":"id,seriesId,title,origin,pubDate,dateUploaded,abstract,resourceMap,beginDate,endDate,read_count_i,geohash_9,datasource,isPublic,documents,northBoundCoord,southBoundCoord,eastBoundCoord,westBoundCoord","start":"0","sort":"dateUploaded desc","facet.mincount":"1","rows":"25","facet":"true","wt":"json","facet.sort":"index"}},"response":{"numFound":2,"start":0,"docs":[{"id":"urn:uuid:1ae34d34-ee3e-441e-a78f-6029320a382c","dateUploaded":"2019-02-22T19:25:57.675Z","datasource":"urn:node:mnTestKNB","isPublic":true,"abstract":"This data package includes a provenance diagram, code, and derived data used in the analysis for the publication titled \"Spatial community structure of groundfish is conserved across the Gulf of Alaska\". In this paper we did the following:  The mechanisms structuring patterns of diversity and community composition can be difficult to identify in large, open ecological systems. However, it is important to understand what drives these patterns at larger scales, especially in the face of climate change and other perturbations. The Gulf of Alaska (GOA) is an ideal study system, because it has complex topography, climate-driven variability, and an on-going groundfish community survey. We used groundfish community data to examine the ecological theory underlying spatial diversity and community composition across 10 study areas in the GOA.  We created geostatistically modeled groundfish abundance and biomass from the Alaska Fisheries Science Center trawl survey data (1984 â€“ 2015) to address inherent errors in trawl observations. We found that species richness, and alpha, beta, and functional diversity varied little both within and between study areas. However, turnover in community composition was significant along a longitudinal gradient, with differences driven by lower-abundance species. Fishing pressure had non-linear effects only on species richness and local diversity, while productivity was linearly related to beta diversity. We conclude that spatial patterns of diversity were not driven by disturbance, but were largely driven by environmental heterogeneity, because of the longitudinal turnover in community composition and high beta diversity (low saturation). In addition, the invariant functional diversity but varying community composition together indicate functional redundancy in this ecosystem. Finally, the spatially invariant alpha and functional diversity show that the underlying community structure of the GOA groundfish community was conserved across this large marine ecosystem.","title":"Code and data for biodiversity, spatial community structure, and spatial community composition of groundfish in the Gulf of Alaska","southBoundCoord":54.0,"northBoundCoord":62.0,"westBoundCoord":-165.0,"eastBoundCoord":-142.0,"beginDate":"1984-01-01T08:00:00Z","endDate":"2015-01-01T08:00:00Z","pubDate":"2019-01-01T08:00:00Z","origin":["Rachael Blake","Collette Ward","Mary Hunsicker","A. Ole Shelton","Anne Hollowed"],"geohash_9":["bd6cgn8dq"],"documents":["urn:uuid:5c284d7b-43f1-41b8-853f-612bb00b85d8","urn:uuid:ca158497-d085-47b5-a584-876c0aff0e15","urn:uuid:9a437585-a27a-4573-8ffa-d63f84f869e1","urn:uuid:1848b6d2-a62e-4e38-acb4-514fe89d0668","urn:uuid:8258aa3d-f07a-4f65-9615-bd8ff296776f","urn:uuid:365489b7-412e-42bc-b4d1-da2d6341cc73","urn:uuid:a985cd68-d55b-46ff-96d6-462a2d0d45d9","urn:uuid:6d909353-0088-4192-a829-cc1a72179273","urn:uuid:4d52000c-68ff-4c67-8585-fbff30191ec6","urn:uuid:687d4638-ebff-48aa-8da1-79dfc25cff8f","urn:uuid:29314a21-697d-4d78-8e5a-a084918f9978","urn:uuid:e8090890-922a-4b15-9558-ceb5020b0eae","urn:uuid:6f3057b7-75ef-41f6-a94b-4e04c62af74a","urn:uuid:652f5482-0d93-47d6-89c3-391d2c6dd26c","urn:uuid:d7a5df22-76c5-49db-b456-d82b7eed1af3","urn:uuid:ab68b2b0-edb4-47bc-b66f-be89d7725165","urn:uuid:a719163e-bef1-4647-9a41-f96a4bba33d9","urn:uuid:3de1af19-36b2-47d2-a44d-6cdd8a1f4e6b","urn:uuid:12e356ae-f380-4e5e-80a0-c60bf12c7fae","urn:uuid:1ae34d34-ee3e-441e-a78f-6029320a382c","urn:uuid:51be3989-c8e9-4108-a8e3-1624d440dd0d","urn:uuid:9260df8c-f0d5-4779-98e5-6dd1a16c62a6","urn:uuid:db0275fe-e0ea-4144-82a9-754bd802020e","urn:uuid:3df2d72d-b429-42e0-b360-342b7d509810","urn:uuid:fe39898c-6ec6-40f1-a978-af0bb9d46d03","urn:uuid:8818def1-6833-470f-9a28-c5f6d8bf56c3","urn:uuid:ff0a4021-103d-44ab-8204-f02c20d17f2f","urn:uuid:a27a187f-ad82-4dd9-a9b0-ae4a10fab123","urn:uuid:f2a1a866-e9a5-4cb5-b7f3-9d8ff1f063c7","urn:uuid:11e7a60c-0507-44e8-b74c-41cc18f2f650","urn:uuid:9a3412d8-9b6d-4047-9c25-bea66d9b8328","urn:uuid:a6f1e615-4aa3-421f-9ed3-9a7af630ea97","urn:uuid:ea1d89d1-c2df-4320-950e-4baf3f9b315a","urn:uuid:41805c07-5654-48f8-b053-1adbfc6f8e81","urn:uuid:1f536fbd-2f92-4a3a-934a-52dcfa71ddb9"],"resourceMap":["resource_map_urn:uuid:da4c144c-9e05-49f9-97a6-39a1b9a9c392","resource_map_urn:uuid:1ae34d34-ee3e-441e-a78f-6029320a382c","resource_map_urn:uuid:a33fff57-a170-4762-b661-cba097ff3feb"],"read_count_i":57},{"id":"urn:uuid:1895dd60-a883-472a-8bea-e8c40896daed","dateUploaded":"2019-01-04T20:12:36.82Z","datasource":"urn:node:mnTestKNB","isPublic":true,"abstract":"This data package includes a provenance diagram, code, and derived data used in the analysis for the publication titled \"Spatial community structure of groundfish is conserved across the Gulf of Alaska\". In this paper we did the following: The mechanisms structuring patterns of diversity and community composition can be difficult to identify in large, open ecological systems. However, it is important to understand what drives these patterns at larger scales, especially in the face of climate change and other perturbations. The Gulf of Alaska (GOA) is an ideal study system, because it has complex topography, climate-driven variability, and an on-going groundfish community survey. We used groundfish community data to examine the ecological theory underlying spatial diversity and community composition across 10 study areas in the GOA.  We created geostatistically modeled groundfish abundance and biomass from the Alaska Fisheries Science Center trawl survey data (1984 â€“ 2015) to address inherent errors in trawl observations. We found that species richness, and alpha, beta, and functional diversity varied little both within and between study areas. However, turnover in community composition was significant along a longitudinal gradient, with differences driven by lower-abundance species. Fishing pressure had non-linear effects only on species richness and local diversity, while productivity was linearly related to beta diversity. We conclude that spatial patterns of diversity were not driven by disturbance, but were largely driven by environmental heterogeneity, because of the longitudinal turnover in community composition and high beta diversity (low saturation). In addition, the invariant functional diversity but varying community composition together indicate functional redundancy in this ecosystem. Finally, the spatially invariant alpha and functional diversity show that the underlying community structure of the GOA groundfish community was conserved across this large marine ecosystem.","title":"Code and data for biodiversity, spatial community structure, and spatial community composition of groundfish in the Gulf of Alaska","southBoundCoord":54.0,"northBoundCoord":62.0,"westBoundCoord":-165.0,"eastBoundCoord":-142.0,"beginDate":"1984-01-01T08:00:00Z","endDate":"2015-01-01T08:00:00Z","pubDate":"2019-01-01T08:00:00Z","origin":["Rachael Blake","Collette Ward","Mary Hunsicker","A. Ole Shelton","Anne Hollowed"],"geohash_9":["bd6cgn8dq"],"documents":["urn:uuid:51473f73-551d-497e-b3c3-158146e546e7","urn:uuid:2ea88930-2d88-4b24-ab06-a3af433d82d9","urn:uuid:d4a1757b-c5ad-44ff-aec9-3ee679d484ef","urn:uuid:4ae25ff8-55ec-4476-9d55-e3f72169b16e","urn:uuid:7434ccf6-1a53-45da-b2b3-4e247049a93c","urn:uuid:3ffa4152-dfe0-4a30-89cf-51a5205e31cc","urn:uuid:cff6449c-74f1-4ce8-bbbb-322f5dd41688","urn:uuid:4479b6cf-25fd-48ff-bff1-aff1dc674073","urn:uuid:cf0dde46-78dc-47ba-8578-65e9c415e0e5","urn:uuid:ea787b1c-6fb3-46a6-b8a7-c09253a3277f","urn:uuid:a89e5b91-8f3e-4fee-a1b1-3c151e568d74","urn:uuid:b2012c2a-762f-4fdb-9e24-635d50be4a10","urn:uuid:0a4dadbe-6c52-4cb1-bc39-b59c58330909","urn:uuid:46f55dfc-09ab-47ea-bc79-fd3440596d20","urn:uuid:a95825b3-c4aa-4d6e-bc26-c24b6f1905d2","urn:uuid:c9374df3-2ede-45a2-8861-958873ab9774","urn:uuid:f08e027f-cbe9-44b9-83e2-ede4d7c3662b","urn:uuid:20be2893-ca78-466c-96d2-1f7aba1d6d2c","urn:uuid:7f5ed733-4548-4f0e-bd29-651d4c8c444d","urn:uuid:7a758d04-4e1b-4488-8052-7c7c8172b878","urn:uuid:de1d1383-e7c6-4fa6-b4d0-b47f1246f55a","urn:uuid:c567b550-ccd9-4a8b-91aa-6e4abe9d2932","urn:uuid:be68dd5e-a919-4dd6-80ee-5c17f51cd068","urn:uuid:61711fb4-bc03-4cfb-bda3-be24d25a8302","urn:uuid:3c86525d-ea7e-423c-bd36-a3019e0780a9","urn:uuid:8292fb2a-c7ac-4c00-acdc-90d6855c8756","urn:uuid:09480f8c-a0e0-4a77-826a-ff736bb5623f","urn:uuid:ffe80796-a7c0-487f-aa3f-ea2beb546e28","urn:uuid:1895dd60-a883-472a-8bea-e8c40896daed","urn:uuid:d83139c6-489e-4f78-8152-d1155a49a21e","urn:uuid:f37799fd-3c2f-4fff-abd5-91fc873891a1","urn:uuid:ff7da56b-c960-4d01-a2cf-b5b87507adc4","urn:uuid:4a0df739-8061-45b5-bb73-27bbd56888e8"],"resourceMap":["urn:uuid:41ec5feb-2a00-4ac3-8ee2-970cb7f28789"],"read_count_i":59}]},"facet_counts":{"facet_queries":{},"facet_fields":{"documents":["urn:uuid:09480f8c-a0e0-4a77-826a-ff736bb5623f",1,"urn:uuid:0a4dadbe-6c52-4cb1-bc39-b59c58330909",1,"urn:uuid:11e7a60c-0507-44e8-b74c-41cc18f2f650",1,"urn:uuid:12e356ae-f380-4e5e-80a0-c60bf12c7fae",1,"urn:uuid:1848b6d2-a62e-4e38-acb4-514fe89d0668",1,"urn:uuid:1895dd60-a883-472a-8bea-e8c40896daed",1,"urn:uuid:1ae34d34-ee3e-441e-a78f-6029320a382c",1,"urn:uuid:1f536fbd-2f92-4a3a-934a-52dcfa71ddb9",1,"urn:uuid:20be2893-ca78-466c-96d2-1f7aba1d6d2c",1,"urn:uuid:29314a21-697d-4d78-8e5a-a084918f9978",1,"urn:uuid:2ea88930-2d88-4b24-ab06-a3af433d82d9",1,"urn:uuid:365489b7-412e-42bc-b4d1-da2d6341cc73",1,"urn:uuid:3c86525d-ea7e-423c-bd36-a3019e0780a9",1,"urn:uuid:3de1af19-36b2-47d2-a44d-6cdd8a1f4e6b",1,"urn:uuid:3df2d72d-b429-42e0-b360-342b7d509810",1,"urn:uuid:3ffa4152-dfe0-4a30-89cf-51a5205e31cc",1,"urn:uuid:41805c07-5654-48f8-b053-1adbfc6f8e81",1,"urn:uuid:4479b6cf-25fd-48ff-bff1-aff1dc674073",1,"urn:uuid:46f55dfc-09ab-47ea-bc79-fd3440596d20",1,"urn:uuid:4a0df739-8061-45b5-bb73-27bbd56888e8",1,"urn:uuid:4ae25ff8-55ec-4476-9d55-e3f72169b16e",1,"urn:uuid:4d52000c-68ff-4c67-8585-fbff30191ec6",1,"urn:uuid:51473f73-551d-497e-b3c3-158146e546e7",1,"urn:uuid:51be3989-c8e9-4108-a8e3-1624d440dd0d",1,"urn:uuid:5c284d7b-43f1-41b8-853f-612bb00b85d8",1,"urn:uuid:61711fb4-bc03-4cfb-bda3-be24d25a8302",1,"urn:uuid:652f5482-0d93-47d6-89c3-391d2c6dd26c",1,"urn:uuid:687d4638-ebff-48aa-8da1-79dfc25cff8f",1,"urn:uuid:6d909353-0088-4192-a829-cc1a72179273",1,"urn:uuid:6f3057b7-75ef-41f6-a94b-4e04c62af74a",1,"urn:uuid:7434ccf6-1a53-45da-b2b3-4e247049a93c",1,"urn:uuid:7a758d04-4e1b-4488-8052-7c7c8172b878",1,"urn:uuid:7f5ed733-4548-4f0e-bd29-651d4c8c444d",1,"urn:uuid:8258aa3d-f07a-4f65-9615-bd8ff296776f",1,"urn:uuid:8292fb2a-c7ac-4c00-acdc-90d6855c8756",1,"urn:uuid:8818def1-6833-470f-9a28-c5f6d8bf56c3",1,"urn:uuid:9260df8c-f0d5-4779-98e5-6dd1a16c62a6",1,"urn:uuid:9a3412d8-9b6d-4047-9c25-bea66d9b8328",1,"urn:uuid:9a437585-a27a-4573-8ffa-d63f84f869e1",1,"urn:uuid:a27a187f-ad82-4dd9-a9b0-ae4a10fab123",1,"urn:uuid:a6f1e615-4aa3-421f-9ed3-9a7af630ea97",1,"urn:uuid:a719163e-bef1-4647-9a41-f96a4bba33d9",1,"urn:uuid:a89e5b91-8f3e-4fee-a1b1-3c151e568d74",1,"urn:uuid:a95825b3-c4aa-4d6e-bc26-c24b6f1905d2",1,"urn:uuid:a985cd68-d55b-46ff-96d6-462a2d0d45d9",1,"urn:uuid:ab68b2b0-edb4-47bc-b66f-be89d7725165",1,"urn:uuid:b2012c2a-762f-4fdb-9e24-635d50be4a10",1,"urn:uuid:be68dd5e-a919-4dd6-80ee-5c17f51cd068",1,"urn:uuid:c567b550-ccd9-4a8b-91aa-6e4abe9d2932",1,"urn:uuid:c9374df3-2ede-45a2-8861-958873ab9774",1,"urn:uuid:ca158497-d085-47b5-a584-876c0aff0e15",1,"urn:uuid:cf0dde46-78dc-47ba-8578-65e9c415e0e5",1,"urn:uuid:cff6449c-74f1-4ce8-bbbb-322f5dd41688",1,"urn:uuid:d4a1757b-c5ad-44ff-aec9-3ee679d484ef",1,"urn:uuid:d7a5df22-76c5-49db-b456-d82b7eed1af3",1,"urn:uuid:d83139c6-489e-4f78-8152-d1155a49a21e",1,"urn:uuid:db0275fe-e0ea-4144-82a9-754bd802020e",1,"urn:uuid:de1d1383-e7c6-4fa6-b4d0-b47f1246f55a",1,"urn:uuid:e8090890-922a-4b15-9558-ceb5020b0eae",1,"urn:uuid:ea1d89d1-c2df-4320-950e-4baf3f9b315a",1,"urn:uuid:ea787b1c-6fb3-46a6-b8a7-c09253a3277f",1,"urn:uuid:f08e027f-cbe9-44b9-83e2-ede4d7c3662b",1,"urn:uuid:f2a1a866-e9a5-4cb5-b7f3-9d8ff1f063c7",1,"urn:uuid:f37799fd-3c2f-4fff-abd5-91fc873891a1",1,"urn:uuid:fe39898c-6ec6-40f1-a978-af0bb9d46d03",1,"urn:uuid:ff0a4021-103d-44ab-8204-f02c20d17f2f",1,"urn:uuid:ff7da56b-c960-4d01-a2cf-b5b87507adc4",1,"urn:uuid:ffe80796-a7c0-487f-aa3f-ea2beb546e28",1],"id":["urn:uuid:1895dd60-a883-472a-8bea-e8c40896daed",1,"urn:uuid:1ae34d34-ee3e-441e-a78f-6029320a382c",1],"geohash_2":["
```

The `CNIdentity.getSubjectInfo()` service could be used  to find the groups that the rightsholder of the portal document belongs to and include those in the query.

Is the rightsholder of the portal document the correct identity to base the filter query on?

**Comment by laurenwalker** [comment]
I like this approach better than using the MN admin identity, since that may include quality scores for datasets the portal owner will never know exist (so there's no ay to exclude them from the portal, if s/he wanted to). 

With this approach, we will still have an issue where people other than the owner viewing the portal may see scores for private datasets, but as we discussed, I don't think there is a ay around that.

**Comment by gothub** [comment]
Implemented in commit f7c3ffe13517672d0ec6488569ea92c041619a8f
